### ISSUE TITLE
[BUG]: 'StepSN' object has no attribute 'binary'

### ISSUE THREAD

**Issue by rprempeh** [POSYDON-code/POSYDON-public#557]
**Describe the bug**
Encountering an error deriving from line 760 of step_SN.py where it reads "set_binary_to_failed(self.binary)". There is no binary attribute defined in the file.

**To Reproduce**
Error arises when attempting to run populations with mechanism set to either 'Fryer+12-rapid' or 'Fryer+12-delayed' following the MODEL02 and MODEL03 descriptions in MODELS.py.
The following are the blocks changed in the .ini file for both population runs leading to the error:

```
[step_SN] 
  import = ['posydon.binary_evol.SN.step_SN', 'StepSN']
  absolute_import = None
  mechanism = 'Fryer+12-rapid'
  engine = None
  PISN = "Marchant+19"
  ECSN = "Podsiadlowski+04"
  conserve_hydrogen_envelope = False
  max_neutrino_mass_loss = 0.5
  max_NS_mass = 2.5
  use_interp_values = False
  use_profiles = True
  use_core_masses = False
  approx_at_he_depletion = False
  kick = True
  kick_normalisation = 'one_over_mass'
  sigma_kick_CCSN_NS = 265.0
  sigma_kick_CCSN_BH = 265.0
  sigma_kick_ECSN = 20.0
  verbose = False
```
```
[step_SN]
  import = ['posydon.binary_evol.SN.step_SN', 'StepSN']
  absolute_import = None
  mechanism = 'Fryer+12-delayed'
  engine = None
  PISN = "Marchant+19"
  ECSN = "Podsiadlowski+04"
  conserve_hydrogen_envelope = False
  max_neutrino_mass_loss = 0.5
  max_NS_mass = 2.5
  use_interp_values = False
  use_profiles = True
  use_core_masses = False
  approx_at_he_depletion = False
  kick = True
  kick_normalisation = 'one_over_mass'
  sigma_kick_CCSN_NS = 265.0
  sigma_kick_CCSN_BH = 265.0
  sigma_kick_ECSN = 20.0
  verbose = False
```

**POSYDON Version:**
Current development version.

**System Configuration (please complete the following information):**
- HPC scheduler: SLURM

**Comment by mkruckow** [comment]
Could you please add the error message?

Otherwise please use
`engine = ''`
as we changed the instructions on the population_params_default.ini recently, see PR #544 .

If there are still issues, please reopen it.
